openmetadata-service/src/main/java/org/openmetadata/service/resources/permissions/PermissionsResource.java

```diff
@@ -81,7 +81,8 @@ public ResultList<ResourcePermission> getPermissions(
               schema = @Schema(type = "string", example = "john"))
           @QueryParam("user")
           String user) {
-    return new ResourcePermissionList(authorizer.listPermissions(securityContext, user));
+    return new ResourcePermissionList(
+        PolicyEvaluator.trimResourcePermissions(authorizer.listPermissions(securityContext, user)));
```

Comment:
Move the trim function call to listPermissions() implementation instead of keeping it in the Resource class.